src/components/proxy-middleware/rule_action_processor/processors/helpers/redirectHelper.js

```diff
@@ -0,0 +1,119 @@
+const parser = require("ua-parser-js");
+import * as Sentry from "@sentry/browser";
+import https from "https"
+
+async function makeSameRequestToDifferentUrl(ctx, url) {
+  try {
+    const requestOptions = {
+      headers:{
+        ...ctx?.clientToProxyRequest?.headers,
+        "Cache-Control": "no-cache",
+      },
+      rejectUnauthorized: false,
+      requestCert: true,
+      agent: false,
+      strictSSL: false,
+    }
+    const {data, response} = await new Promise ((resolve, reject) => {
+      let request = https.get(url, requestOptions, (res) => {
+        const dataBuffers = []
+        res.on('data', (buffer) => {
+          dataBuffers.push(buffer)
+        });
+        res.on('end', () => {
+          resolve({data: Buffer.concat(dataBuffers).toString(), response: res})
+        });
+      })
+      request.on('error', (error) => {
+        console.error(error);
+        reject(error)
+      });
+    })
+
+    return {success: true, response, data}
+  } catch (error) {
+    Sentry.captureException(error);
+    console.error(error)
+    return {success: false, error}
+  }
+}
+
+const willCreateMixedResponse = (ctx , destinationUrl) => {
+  let user_agent_str = null;
+  user_agent_str = ctx?.clientToProxyRequest?.headers["user-agent"];
+  console.log("handling mixed response. i got headers",ctx?.clientToProxyRequest?.headers)
+  const user_agent = parser(user_agent_str)?.browser?.name;
+  const LOCAL_DOMAINS = ["localhost", "127.0.0.1"];
+
+  return ctx.isSSL && (
+    user_agent === "Safari" ||
+    !LOCAL_DOMAINS.some((domain) => destinationUrl.includes(domain))
+  )
+}
+
+export const handleMixedResponse = async (ctx, destinationUrl) => {
```

Comment:
handleMixedResponse & handleServerSideRedirect can be a single function as both are doing same thing. We can to move the conditionals to the calling method